src/signtool/IBM_Container.h

```diff
@@ -0,0 +1,253 @@
+/* IBM_PROLOG_BEGIN_TAG                                                   */
+/* This is an automatically generated prolog.                             */
+/*                                                                        */
+/* $Source: src/signtool/IBM_Container.h $                                */
+/*                                                                        */
+/* OpenPOWER sb-signing-utils Project                                     */
+/*                                                                        */
+/* Contributors Listed Below - COPYRIGHT 2016                             */
+/* [+] International Business Machines Corp.                              */
+/*                                                                        */
+/*                                                                        */
+/* Licensed under the Apache License, Version 2.0 (the "License");        */
+/* you may not use this file except in compliance with the License.       */
+/* You may obtain a copy of the License at                                */
+/*                                                                        */
+/*     http://www.apache.org/licenses/LICENSE-2.0                         */
+/*                                                                        */
+/* Unless required by applicable law or agreed to in writing, software    */
+/* distributed under the License is distributed on an "AS IS" BASIS,      */
+/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or        */
+/* implied. See the License for the specific language governing           */
+/* permissions and limitations under the License.                         */
+/*                                                                        */
+/* IBM_PROLOG_END_TAG                                                     */
+
+#ifndef __IBM_CONTAINER_H_
+#define __IBM_CONTAINER_H_
+
+#include <stdint.h>
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "IBM_Utils.h"
+
+struct ContainerHdr
+{
+    uint32_t  m_magicNumber;                    // (17082011)
+    uint16_t  m_version;                        // (1: see versions above)
+    uint64_t  m_containerSize;                  // filled by caller
+    uint64_t  m_targetHrmor;                    // filled by caller
+    uint64_t  m_stackPointer;                   // filled by caller
+    uint8_t   m_hwPkeyA[ECDSA521_KEY_SIZE];
+    uint8_t   m_hwPkeyB[ECDSA521_KEY_SIZE];
+    uint8_t   m_hwPkeyC[ECDSA521_KEY_SIZE];
+
+    ContainerHdr();
+
+    void PrintHeader() const;
+    void GetHeaderBytes( std::vector<uint8_t>& packet ) const;
+
+}  __attribute__ ((packed));
```

Comment:
What is the reason for these to be packed?  Are you doing a raw memcpy of the full object?